src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx

```diff
@@ -120,19 +121,39 @@ export class CreateDataSourceForm extends React.Component<
   };
 
   onChangeAuthType = (value: string) => {
-    const valueToSave =
-      value === AuthType.UsernamePasswordType ? AuthType.UsernamePasswordType : AuthType.NoAuth;
-
+    const valueToSave = value as AuthType;
     const formErrorsByField = {
       ...this.state.formErrorsByField,
       createCredential: { ...this.state.formErrorsByField.createCredential },
+      awsCredential: { ...this.state.formErrorsByField.awsCredential },
     };
     if (valueToSave === AuthType.NoAuth) {
       formErrorsByField.createCredential = {
         username: [],
         password: [],
       };
+      formErrorsByField.awsCredential = {
+        region: [],
+        accessKey: [],
+        secretKey: [],
+      };
+    }
+
+    if (valueToSave === AuthType.UsernamePasswordType) {
```

Comment:
142 and 150 should be swapped?

Reply:
Removed this whole block of resetting formErrors based on type. I don't think it makes sense. Input rrrors should be preserved when toggling between auth types, because the inputs is also preserved